### PR TITLE
Client properties

### PR DESCRIPTION
**Pull request by jburel**
corresponding PR of #4474 for insight
Add support for new properties.

Testing: 
 * Follow test used for #4474

cc @pwalczysko @aleksandra-tarkowska

### CODE REVIEW

**Comment by pwalczysko** [comment]
After discussion with @jburel :
To test
  - orphaned 
  - group owner name manipulation, i.e. alias properties (``leader``, ``colleague`` etc)

**Comment by pwalczysko** [comment]
Orpahned folder testing:
There is a discrepancy in behaviour between Web and Insight with respect to the Orphaned folder visibility by other members of group when ``omero.client.ui.tree.orphans.enabled false`` is set.
   - in Web, member cannot see Orphan folder of another member at all
   - in Insight, member can see Orphaned folder of another member, it just shows ``Empty`` label even when images are in it (this is expected) -> the discrepancy is in the fact that Insight still shows the empty folder

In this screenshot, user-2 was logged in, user-1 has 2 images in Orphaned.
screen shot 2016-03-03 at 16 34 57

**Comment by pwalczysko** [comment]
Bug: 
  - discrepancy between Web and insight when I disable the ``colleagues`` label
  - Insight seems to be in the wrong here (showing strangely greyed out ``admin1`` which is in the group as a member (not group owner and also showing the members, although ``colleagues`` label is ``false``)

screen shot 2016-03-03 at 16 54 21
screen shot 2016-03-03 at 16 54 15
screen shot 2016-03-03 at 16 53 58
screen shot 2016-03-03 at 16 53 50

**Comment by pwalczysko** [comment]
After I have set 
```
bin/omero config get
omero.client.ui.menu.dropdown.colleagues.enabled=true
omero.client.ui.menu.dropdown.everyone.enabled=true
omero.client.ui.menu.dropdown.everyone.label=Rank and file
omero.client.ui.menu.dropdown.leaders.enabled=false
omero.client.ui.menu.dropdown.leaders.label=Big bosses
```
I have restarted server. After server restart, I have restarted Insight. I get an error (see below) followed by full crash immediately after login.
```
java.lang.Exception: Abnormal termination due to an uncaught exception.
java.lang.ArrayIndexOutOfBoundsException: 1 >= 1
	at java.util.Vector.elementAt(Vector.java:474)
	at javax.swing.tree.VariableHeightLayoutCache.getNode(VariableHeightLayoutCache.java:994)
	at javax.swing.tree.VariableHeightLayoutCache.getMaxNodeWidth(VariableHeightLayoutCache.java:1007)
	at javax.swing.tree.VariableHeightLayoutCache.getPreferredWidth(VariableHeightLayoutCache.java:309)
	at javax.swing.plaf.basic.BasicTreeUI.updateCachedPreferredSize(BasicTreeUI.java:1900)
	at javax.swing.plaf.basic.BasicTreeUI.getPreferredSize(BasicTreeUI.java:2003)
	at javax.swing.plaf.basic.BasicTreeUI.getPreferredSize(BasicTreeUI.java:1991)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1662)
	at javax.swing.JTree.getPreferredScrollableViewportSize(JTree.java:3407)
	at javax.swing.ViewportLayout.preferredLayoutSize(ViewportLayout.java:93)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at javax.swing.ScrollPaneLayout.preferredLayoutSize(ScrollPaneLayout.java:492)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:719)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at info.clearthought.layout.TableLayout.calculateLayoutSize(Unknown Source)
	at info.clearthought.layout.TableLayout.preferredLayoutSize(Unknown Source)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at org.jdesktop.swingx.JXCollapsiblePane.getPreferredSize(JXCollapsiblePane.java:619)
	at org.jdesktop.swingx.plaf.basic.BasicTaskPaneUI.getPreferredSize(BasicTaskPaneUI.java:252)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1662)
	at java.awt.GridBagLayout.GetLayoutInfo(GridBagLayout.java:1115)
	at java.awt.GridBagLayout.getLayoutInfo(GridBagLayout.java:916)
	at java.awt.GridBagLayout.preferredLayoutSize(GridBagLayout.java:736)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at javax.swing.ScrollPaneLayout.layoutContainer(ScrollPaneLayout.java:791)
	at java.awt.Container.layout(Container.java:1510)
	at java.awt.Container.doLayout(Container.java:1499)
	at java.awt.Container.validateTree(Container.java:1695)
	at java.awt.Container.validate(Container.java:1630)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:711)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.validateInvalidComponents(RepaintManager.java:708)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1731)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Abnormal termination due to an uncaught exception.
java.lang.ArrayIndexOutOfBoundsException: 1 >= 1
	at java.util.Vector.elementAt(Vector.java:474)
	at javax.swing.tree.VariableHeightLayoutCache.getNode(VariableHeightLayoutCache.java:994)
	at javax.swing.tree.VariableHeightLayoutCache.getMaxNodeWidth(VariableHeightLayoutCache.java:1007)
	at javax.swing.tree.VariableHeightLayoutCache.getPreferredWidth(VariableHeightLayoutCache.java:309)
	at javax.swing.plaf.basic.BasicTreeUI.updateCachedPreferredSize(BasicTreeUI.java:1900)
	at javax.swing.plaf.basic.BasicTreeUI.getPreferredSize(BasicTreeUI.java:2003)
	at javax.swing.plaf.basic.BasicTreeUI.getPreferredSize(BasicTreeUI.java:1991)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1662)
	at javax.swing.JTree.getPreferredScrollableViewportSize(JTree.java:3407)
	at javax.swing.ViewportLayout.preferredLayoutSize(ViewportLayout.java:93)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at javax.swing.ScrollPaneLayout.preferredLayoutSize(ScrollPaneLayout.java:492)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at java.awt.BorderLayout.preferredLayoutSize(BorderLayout.java:719)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at info.clearthought.layout.TableLayout.calculateLayoutSize(Unknown Source)
	at info.clearthought.layout.TableLayout.preferredLayoutSize(Unknown Source)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at org.jdesktop.swingx.JXCollapsiblePane.getPreferredSize(JXCollapsiblePane.java:619)
	at org.jdesktop.swingx.plaf.basic.BasicTaskPaneUI.getPreferredSize(BasicTaskPaneUI.java:252)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1662)
	at java.awt.GridBagLayout.GetLayoutInfo(GridBagLayout.java:1115)
	at java.awt.GridBagLayout.getLayoutInfo(GridBagLayout.java:916)
	at java.awt.GridBagLayout.preferredLayoutSize(GridBagLayout.java:736)
	at java.awt.Container.preferredSize(Container.java:1796)
	at java.awt.Container.getPreferredSize(Container.java:1780)
	at javax.swing.JComponent.getPreferredSize(JComponent.java:1664)
	at javax.swing.ScrollPaneLayout.layoutContainer(ScrollPaneLayout.java:791)
	at java.awt.Container.layout(Container.java:1510)
	at java.awt.Container.doLayout(Container.java:1499)
	at java.awt.Container.validateTree(Container.java:1695)
	at java.awt.Container.validate(Container.java:1630)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:711)
	at javax.swing.RepaintManager$3.run(RepaintManager.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at javax.swing.RepaintManager.validateInvalidComponents(RepaintManager.java:708)
	at javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1731)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Exception in thread "AWT-EventQueue-0"

	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.showErrorDialog(UserNotifierImpl.java:189)
	at org.openmicroscopy.shoola.env.ui.UserNotifierImpl.notifyError(UserNotifierImpl.java:289)
	at org.openmicroscopy.shoola.env.AbnormalExitHandler.doTermination(AbnormalExitHandler.java:147)
	at org.openmicroscopy.shoola.env.AbnormalExitHandler.terminate(AbnormalExitHandler.java:85)
	at org.openmicroscopy.shoola.env.RootThreadGroup.uncaughtException(RootThreadGroup.java:69)
	at java.awt.EventDispatchThread.processException(EventDispatchThread.java:223)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:215)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by pwalczysko** [comment]
I can see some commits addressing the two issues https://github.com/openmicroscopy/openmicroscopy/pull/4507#issuecomment-191844851 and https://github.com/openmicroscopy/openmicroscopy/pull/4507#issuecomment-191864020. 
I can check them on Tuesday.

Another serious problem appeared though https://github.com/openmicroscopy/openmicroscopy/pull/4507#issuecomment-191971840
Will stop testing here for now.
Can continue on Tuesday if convenient.

**Comment by jburel** [comment]
@pwalczysko: I don't the crash is related to the changes. 
We have seen that before but it is difficult to repeat.
We can review when you get back, no rush

**Comment by dominikl** [comment]
Code changes look good, no functional testing done (leave that to @pwalczysko again).

**Comment by pwalczysko** [comment]
Trying to repeat the bug https://github.com/openmicroscopy/openmicroscopy/pull/4507#issuecomment-191864020 - exactly the same setup, then tried to click and unclick the ``Rank and file`` menu item (the ``Rank and file``, in this config meaning ``All members`` did not display the data in Insight, it did in Web). After that, in Insight, I unclicked the ``Display users`` checkbox as well. 
Error occurred.

```
java.lang.Exception: Abnormal termination due to an uncaught exception.
java.lang.NullPointerException
   at org.openmicroscopy.shoola.agents.treeviewer.browser.BrowserUI.buildOrphanImagesNode(BrowserUI.java:1223)
   at org.openmicroscopy.shoola.agents.treeviewer.browser.BrowserUI.setExperimenterData(BrowserUI.java:1915)
   at org.openmicroscopy.shoola.agents.treeviewer.browser.BrowserComponent.setExperimenterData(BrowserComponent.java:1188).....
```
and full crash of Insight.

**Comment by mtbc** [comment]
@jburel: Needs relisting?

**Comment by jburel** [comment]
@mtbc: not yet.

**Comment by jburel** [comment]
@pwalczysko:re-listing the PR the problem fixed by the last commit might have led to an unstable situation.

**Comment by pwalczysko** [comment]
Bug: The ``All members`` checkbox does not select all the checkboxes at the members names.
Insight latest
screen shot 2016-03-15 at 11 45 35

Insight merge
screen shot 2016-03-15 at 11 48 19

**Comment by pwalczysko** [comment]
All the other functionalities and retests passed, but the bug https://github.com/openmicroscopy/openmicroscopy/pull/4507#issuecomment-196784302 seems like unwanted regression.

**Comment by jburel** [comment]
@pwalczysko: The "All members" is now the default so coming from server so both insight and wed are unified.

File: components/insight/SRC/org/openmicroscopy/shoola/agents/treeviewer/view/ToolBar.java [hunk 1]
```diff
@@ -85,6 +84,7 @@
 import org.openmicroscopy.shoola.env.LookupNames;
 import org.openmicroscopy.shoola.env.data.model.ScriptObject;
 import org.openmicroscopy.shoola.env.ui.TaskBar;
+import org.openmicroscopy.shoola.util.CommonsLangUtils;
```
Comment (sbesson) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/treeviewer/view/ToolBar.java]: Note this import will conflict with https://github.com/openmicroscopy/openmicroscopy/commit/7db2c2565e0b7094dc47f4febc83eb5280b46ffb#diff-1799b1647ef5bf4722c7f08b7b5c15e2R88 and the conflict will need to be resolved when merging `develop` into `regions` /cc @dominikl @mtbc

Comment (jburel) [inline components/insight/SRC/org/openmicroscopy/shoola/agents/treeviewer/view/ToolBar.java]: It can be removed if it makes things easier

**Comment by pwalczysko** [comment]
Tested on eel merge. The All members checkbox now selects all members as expected.
Ready to merge.

**Comment by jburel** [comment]
This PR should hopefully no longer conflict with the regions work 
cc @sbesson @dominikl

**Comment by mtbc** [comment]
https://github.com/snoopycrimecop/openmicroscopy/compare/regions/merge/trigger...snoopycrimecop:develop/merge/daily looks good at least.

**Pull request merged**